source/server/hot_restart_impl.cc

```diff
@@ -140,36 +139,10 @@ HotRestartImpl::HotRestartImpl(Options& options)
 
 Stats::RawStatData* HotRestartImpl::alloc(const std::string& name) {
   // Try to find the existing slot in shared memory, otherwise allocate a new one.
-  std::unique_lock<Thread::BasicLockable> lock(stat_lock_);
```

Comment:
The whole functionality of `HotRestartImpl::alloc`, including the lock, got moved up into `BlockRawStatDataAllocator::alloc`. Same for `::free()`.